packages/jq/src/lib/evaluate/filters/builtinNativeFilters.ts

```diff
@@ -435,8 +435,31 @@ export const builtinNativeFilters: Record<string, NativeFilter> = {
     *'tojson/0'() {
       throw notImplementedError('tojson/0');
     },
-    *'tonumber/0'() {
-      throw notImplementedError('tonumber/0');
+    *'tonumber/0'(input: unknown) {
+      const type = typeOf(input);
+      switch (typeOf(input)) {
+        case Type.string: {
```

Comment:
Hello, thanks for the PR. It looks mostly good.

I'd just make a tiny improvement before merging it. It is not necessary to call `typeOf` twice:

```suggestion
    *'tonumber/0'(input: unknown) {
      const type = typeOf(input);
      switch (type) {
        case Type.string: {
```